### ISSUE TITLE
Compatible with Okta Single-page client?

### ISSUE THREAD

**Issue by DJFarr** [IdentityModel/IdentityModel.AspNetCore.OAuth2Introspection#152]
Hi,

Please forgive the newbie question, I'm quite new to OIDC.

I need my .Net API to introspect our front-end's tokens, and we're in the process of migrating to OKTA from Keycloak.
Previously this library appeared to send the Client ID and Client Secret in a Bearer Authorization header, but OKTA expects to receive just the client_id and token_type_hint in an x-www-form-urlencoded body (for Single -Page).

So my question is: does this library support this? I can't find a single online tutorial showing how to use Okta's introspect endpoint in .Net Core and I was hoping this library would still work.

Thank you for your time.

**Comment by leastprivilege** [comment]
This is an impementation of https://datatracker.ietf.org/doc/html/rfc7662

I don't know what Okta is exactly doing. sorry.

**Comment by DJFarr** [comment]
Hi I did manage to get this working.

Our app is a Single-Page app, and while I'm still not sure how to get this working with a single OKTA client in this situation, we did manage to get it working by making 2 (on a custom auth Server).

Our App authenticated via the OKTA Single-Page client, which has no client_secret. It sends requests to our API along with the Token, which gets introspected against a 2nd OKTA client set up with client credentials, used by this library. As both apps are on the same custom auth server, the 2nd is able to introspect tokens issued through the 1st.

This might be common knowledge for some but I thought I'd post here just to share.
